src/Microsoft.OData.Core/ValidationKinds.cs

```diff
@@ -36,6 +36,16 @@ public enum ValidationKinds
         /// </summary>
         ThrowIfTypeConflictsWithMetadata = 4,
 
+        /// <summary>
+        /// Validates that top level properties cannot be null. If a top level property is null,
+        /// the response should be 204 (NoContent) per the spec: 
+        /// 11.2.3 Requesting Individual Properties
+        /// ...
+        /// If the property is single-valued and has the null value, the service responds with 204 No Content.
+        /// ...
+        /// </summary>
+        ThrowOnTopLevelNullProperty = 8,

```

Comment:
It seems that we are turning an error case to a success case, which match the protocol, so we shouldn't need to provide the option of the flag ValidationKinds.ThrowOnTopLevelNullProperty?

Reply:
My read of 11.2.3 is that is a top-level property is all that is to be written, and the value of that property is null, the service should return 204. My thought is the writer could either write nothing to the stream and let the caller determine 204 is appropriate because the stream is empty OR the writer could throw in this case since 204 is appropriate and it's the callers job to return 204. So the default is throw and it's controlled by the validation value ThrowOnTopLevelNullProperty.

For when this causes problems for folks, as in the case of this bug, the options allows ThrowOnTopLevelNullProperty to be turned off, which will allow the writer to write the null value to the stream. This is NOT compliant with protocol. 

So I would describe the fix as formalize the existing error case but allow this behavior to be turned off and get a similar behavior to the 6.x library.